### DISCUSSION TITLE
Comparing yaml files in v4

### DISCUSSION THREAD

**Discussion by ogruene** [mikefarah/yq#673] [Q&A]
Hi all,

in v3 we used the following commands for comparing two yaml files:
```
yq compare -p p file-1.yaml file-2.yaml --indent 2 "**"
yq compare -p pv file-1.yaml file-2.yaml --indent 2 "**"
```

Unfortunately, I could not find a way until now how to do this with v4. Can anybody give some hints?

Thank you :-)

Oliver Grüneberg <<oliver.grueneberg@daimler.com>>, Daimler TSS GmbH
[Legal Imprint](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)

**Comment by mikefarah** [comment]
Can you use `yq` with prettyPrint + sortKeys and then compare the outputs of both files using diff?

**Reply by ogruene** [reply]
I will try this but I'm not sure if this can easily produce a result as it was in v3 ...

**Reply by adusumillipraveen** [reply]
I have hit into same and tried the above approach, problem comes when u have lists like:

```
config:
  - name: value1
     key: value2
  - name: value3
     key: value4
```
 
 
 yq@3 used to handle such comparisons  very well.

**Comment by rdxmb** [comment]
looking for a comparison / diff, too. 

I've found https://mikefarah.gitbook.io/yq/v/v3.x/commands/compare , which does not work with  v4.35.1

```
$ yq compare --help | head -n20
yq is a portable command-line YAML processor (https://github.com/mikefarah/yq/) 
See https://mikefarah.gitbook.io/yq/ for detailed documentation and examples.

## Evaluate Sequence ##
This command iterates over each yaml document from each given file, applies the 
expression and prints the result in sequence.

Usage:
  yq eval [expression] [yaml_file1]... [flags]

Aliases:
  eval, e

Examples:

# Reads field under the given path for each file
yq e '.a.b' f1.yml f2.yml 

# Prints out the file
yq e sample.yaml 
```
